### ISSUE TITLE
Exported function that calls an undeclared function is not caught

### ISSUE THREAD

**Issue by Zenwolf** [eslint/eslint#4397]
**Version:**
1.9.0

**Config:**
```
{
    "env": {
        "browser": true,
        "es6": true,
        "node": true
    },

    "ecmaFeatures": {
        "modules": true
    },

    "rules": {
        "no-unused-vars": [2, { "args": "none" }]
    }
}
```

**Example broken code:**
```
export function foo() {
    return bar();
}
```

**Command line:**
```
eslint --no-eslintrc -c ./eslint.json ./src
```
Where `src/foo.js` has the example code above.

**Result:**
`bar()` is not reported as an undefined function. If I remove the `export` then `bar()` is caught.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by Zenwolf** [comment]
This must have been a recent change since it does not happen in version `1.8.0`.

**Comment by nzakas** [comment]
Could potentially be an escope issue, we will have to double check.

**Comment by ilyavolodin** [comment]
Sorry, but your config file doesn't have `no-undef` in it. Is it just a typo?

**Comment by Zenwolf** [comment]
@ilyavolodin I believe the default has always been to error:
https://github.com/eslint/eslint/blob/master/packages/eslint-config-eslint/default.yml#L62

Also, in version `0.24.1` this error was caught correctly with no specific "no-undef" in the config. I also added it here to see if it made a difference, and it doesn't.

**Comment by IanVS** [comment]
As of version `1.0.0`, there are no rules enabled by default.  Please see http://eslint.org/docs/user-guide/migrating-to-1.0.0 for more information.

**Comment by nzakas** [comment]
@Zenwolf please provide the exact output when you run ESLint.

**Comment by Zenwolf** [comment]
@ilyavolodin @nzakas Ah, I was incorrect in assuming the defaults were still used. This is most likely the problem. I created a new project from scratch and verified the expected and unexpected behavior was due to not explicitly setting the `"no-undef": [2]` rule. There must have been something else interfering with the rule when I tested it in my other project. Glad to hear this is not a bug and thanks for pointing me to the migration guide (which I hadn't seen).
